test/e2e/config/azure-dev.yaml

```diff
@@ -173,10 +173,10 @@ providers:
 
 variables:
   AKS_KUBERNETES_VERSION: "${KUBERNETES_VERSION:-stable-1.19}"
-  KUBERNETES_VERSION: "${KUBERNETES_VERSION:=v1.22.5}"
+  KUBERNETES_VERSION: "${KUBERNETES_VERSION:=stable-1.22}"
   ETCD_VERSION_UPGRADE_TO: "3.5.0-0"
   COREDNS_VERSION_UPGRADE_TO: "v1.8.4"
-  KUBERNETES_VERSION_UPGRADE_TO: "${KUBERNETES_VERSION_UPGRADE_TO:=v1.22.5}"
+  KUBERNETES_VERSION_UPGRADE_TO: "${KUBERNETES_VERSION_UPGRADE_TO:=stable-1.22}"
```

Comment:
Will this select the same version we are already on since both deployment and upgrade would be the same (`stable-1.22`)?

Reply:
upgrade from is set to 1.22.1 so it would upgrade from 1.22.1 to 1.22.6 (or latest)

I believe we had to set ugrade from to 1.22 (used to be 1.21) in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1792, we might be able to open a separate PR to revert it now?

Reply:
ah, missed that was a separate configuration. Thought it was from `KUBERNETES_VERSION`.

We can't revert as we still need 1.22 for windows but we could set `KUBERNETES_VERSION_UPGRADE_TO` to 1.23 now that it is released.

Reply:
I'll open a separate PR to change upgrade, need to change the worker count as well since CAPI 1.1 beta merged